docs/en/custom-elements/introduction.md

```diff
@@ -0,0 +1,164 @@
+# Introduction
+
+Dojo widgets can be compiled into custom elements and therefore be used like any other type of HTML tag. The widget's properties will be automatically converted into HTML attributes, properties, or events.
+
+Each widget is bundled into a JavaScript file and a CSS file, both of which are imported into your HTML.
+
+> index.html
+
+```html
+<html>
+  <head>
+    <link rel="stylesheet" href="ref-email-signup.css">
+  </head>
+  <body>
+    <ref-email-signup></ref-email-signup>
+    <script async src="ref-email-signup.js"></script>
+  </body>
+</html>
+```
+
+| Feature           | Description                                                                                                                                                              |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Widget Properties | Based on the type and name of the property, Dojo will decide if the property should be an attribute, property, or event on the custom element.                           |
+| Child Renderers   | Widgets that use child objects or renderers are supported in custom elements by using slots.                                                                             |
+| Lazy Loading      | The custom element script does not need to be loaded before the custom element is referenced on the page. Elements will be hydrated automatically when the script loads. |
+
+## Building Custom Elements
+
+Dojo widgets can be built as custom elements using @dojo/cli-build-widget. For this example, we're going to use a simple email subscription widget.
+
+> src/widgets/EmailSignup.tsx
```

Comment:
it seems like this example is quite big for the Introduction/Hello World part of a ref guide. For this basic one could we not simply cover the building of a HelloWorld component. We can then delve into the other stuff in the supplemental

Reply:
That's fair. I can move the entire "Using custom elements" contents to the supplemental that shows exactly how props/attributes/events are handled. For the basic one I'll just create a widget that has a single string property that can be used as a hello world.